pkg/tfbridge/provider.go

```diff
@@ -1521,13 +1521,15 @@ func (p *ProviderInfo) SetAutonaming(maxLength int, separator string) {
 	const nameProperty = "name"
 	for resname, res := range p.Resources {
 		if schema := p.P.ResourcesMap().Get(resname); schema != nil {
-			// Only apply auto-name to input properties (Optional || Required) named `name`
-			if sch := schema.Schema().Get(nameProperty); sch != nil && (sch.Optional() || sch.Required()) {
```

Comment:
Could there be any implicit conversions which applied here, which will now break?
Say for fields of type `bytes`, would the auto-naming have worked but would now break?

Reply:
I'd be surprised, but it's worth checking. I'll run through our tier 1 providers and confirm that there are no unwanted changes.

Reply:
I checked:

- gcp: no changes.
- aws: only the one expected change that this is fixing.
- azure: no changes.
- azuread: no changes.